### PR TITLE
[v7r1] SummarizeLogsAgent uses list instead of tuples

### PR DESCRIPTION
**Pull request by chaen**
Based on https://github.com/DIRACGrid/DIRAC/pull/4767

BEGINRELEASENOTES
*RSS
CHANGE: SummarizeLogAgent uses list instead of tuples

*WMS
CHANGE: JobMonitoring uses cls/self attributes instead of global variable

ENDRELEASENOTES

### CODE REVIEW

File: WorkloadManagementSystem/Service/JobMonitoringHandler.py [hunk 1]
```diff
@@ -385,10 +385,10 @@ def export_getJobPageSummaryWeb(self, selectDict, sortList, startItem, maxItems,
       orderAttribute = None
 
     statusDict = {}
-    result = gJobDB.getCounters('Jobs', ['Status'], selectDict,
-                                newer=startDate,
-                                older=endDate,
-                                timeStamp='LastUpdateTime')
+    result = self.gJobDB.getCounters('Jobs', ['Status'], selectDict,
```
Comment (fstagni) [inline WorkloadManagementSystem/Service/JobMonitoringHandler.py:388]: `cls.gJobDB`

File: WorkloadManagementSystem/Service/JobMonitoringHandler.py [hunk 2]
```diff
@@ -407,8 +407,8 @@ def export_getJobPageSummaryWeb(self, selectDict, sortList, startItem, maxItems,
       if iniJob >= nJobs:
         return S_ERROR('Item number out of range')
 
-      result = gJobDB.selectJobs(selectDict, orderAttribute=orderAttribute,
-                                 newer=startDate, older=endDate, limit=(maxItems, iniJob))
+      result = self.gJobDB.selectJobs(selectDict, orderAttribute=orderAttribute,
```
Comment (fstagni) [inline WorkloadManagementSystem/Service/JobMonitoringHandler.py:410]: `cls` not `self` (also below)

File: WorkloadManagementSystem/Service/JobMonitoringHandler.py [hunk 3]
```diff
@@ -499,62 +499,60 @@ def export_getJobStats(attribute, selectDict):
...
+  def export_getJobsPrimarySummary(cls, jobIDs):
+    return cls.getAttributesForJobList(jobIDs, PRIMARY_SUMMARY)
 
 ##############################################################################
   types_getJobParameter = [[basestring, int, long], basestring]
 
-  @staticmethod
   @ignoreEncodeWarning
-  def export_getJobParameter(jobID, parName):
+  def export_getJobParameter(self, jobID, parName):
```
Comment (fstagni) [inline WorkloadManagementSystem/Service/JobMonitoringHandler.py:511]: Not a `@classmethod`?

File: WorkloadManagementSystem/Service/JobMonitoringHandler.py (hunk 1 above)
Comment (chaen) [reply WorkloadManagementSystem/Service/JobMonitoringHandler.py:388]: no it's not a class method. Same for the others

File: WorkloadManagementSystem/Service/JobMonitoringHandler.py (hunk 3 above)
Comment (chaen) [reply WorkloadManagementSystem/Service/JobMonitoringHandler.py:511]: no because it needs the elastic db

**Comment by fstagni** [comment]
Should this still be [WIP]? Anything to add?

**Comment by chaen** [comment]
No I don't think there is anything to add. It just needs your PR, and that you rework the `getAttributesForJobList` or protect it, as you see fit

**Review by fstagni — APPROVED** [review]

**Review by atsareg — APPROVED** [review]

**Pull request merged**